docs/docs/60-new-docs/50-user-guide/20-how-to-guides/16-working-with-warehouses.md

```diff
@@ -0,0 +1,198 @@
+    tagged in this way, including any new or updated digest.
+
+- `Lexical`: This strategy selects the image with the lexicographically greatest tag, making it suitable
+    for scenarios where tags incorporate date/time stamps in formats like `yyyymmdd`. When using this
+    strategy, it's recommended to pair it with a regular expression in the `allowTags` field to limit
+    eligibility to tags that match the expected format, ensuring the correct selection of tags.
+
+- `NewestBuild`: This strategy selects the image with the most recent build time.
+
+    :::warning
+    `NewestBuild` requires retrieving metadata for every eligible tag, which can be slow and is likely to
+    exceed the registry's rate limits. It's advisable to use the `allowTags` field to limit
+    the number of tags for which metadata is retrieved, thereby reducing the risk of hitting rate limits.
+    :::
+
+- **SemVer**: This strategy selects the image that best matches a semantic versioning constraint.
+
+    :::info
+    Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle these contraints,
+    allowing users to define and manage versions precisely.
+    :::
+
+### Git Subscription
+
+In subscriptions to Git repositories, the `commitSelectionStrategy` field
+specifies the method for selecting the desired commit.
+
+The available strategies for subscribing to a Git repository are:
+
+- `Lexical`: Selects the commit referenced by the lexicographically greatest tag.
+    It is particularly useful in scenarios where commit references, such as tags or branches,
+    incorporate date/time stamps in formats like `yyyymmdd`.
+    To ensure the correct selection, it's advisable to use regular expressions in the
+    `allowTags` or `allowBranches` field, which limit the acceptable format of the references,
+    preventing the selection of undesired tags like `zzz-custom` over something like `nightly-20241211`.
+- `NewestFromBranch`: Selects the most recent commit from a specified branch. It's useful when tracking the latest changes in a branch that receives regular updates.
+- `NewestTag`: Selects the most recent commit associated with a tag. Since tags are typically immutable,
+    there should be only one commit per tag.
+    To optimize this strategy, it's recommended to constrain the eligible tags using regular expressions or specific patterns,
+    ensuring the selection is limited to tags that follow a consistent naming convention.
+- `SemVer`: Selects the commit referenced by a *tag* that best matches the constraint.
+
+#### Git Subscription Path Filtering
+
+In some cases, it may be necessary to constrain the paths within a Git
+repository that a `Warehouse` will consider as triggers for `Freight`
+production. This is especially useful for GitOps repositories that are
+"monorepos" containing configuration for multiple applications.
```

Comment:
It would be useful to link to this pattern:

https://docs.kargo.io/new-docs/user-guide/patterns/#monorepo-layout

It would also be useful to add a link to this section in the proper spot in that patterns doc.